CRM/Contribute/Form/Contribution/Main.php

```diff
@@ -141,6 +141,10 @@ public function preProcess() {
       $this->assign('footer_text', $this->_values['footer_text']);
     }
 
+    if (empty($this->_values['custom_pre_id']) && empty($this->_values['custom_post_id'])) {
+      CRM_Core_Error::fatal(ts("Please include a Profile for online registration that contains an Email Address field and / or First Name + Last Name fields."));
```

Comment:
error msg needs to be corrected  - remove "... for online registration ..." since that's specific to events registration.